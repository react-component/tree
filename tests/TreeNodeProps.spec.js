/* eslint-disable no-undef */
import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Tree, { TreeNode } from '..';

/**
 * For refactor purpose. All the props should be passed by test
 */

describe('TreeNode Props', () => {
  // prefixCls - is defined by Tree, TreeNode can not change it
  // expanded - is defined by Tree, TreeNode can not change it

  it('className', () => {
    const wrapper = render(
      <Tree defaultExpandAll>
        <TreeNode key="0-0">
          <TreeNode key="0-0-0" className="tree-node-cls">
            <TreeNode />
          </TreeNode>
          <TreeNode key="0-0-1" />
        </TreeNode>
        <TreeNode key="0-1" />
      </Tree>
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  // disabled - is already full test in Tree.spec.js
  // disableCheckbox - is already full test in Tree.spec.js
  // title - is already full test in Tree.spec.js
  // key - is already full test in Tree.spec.js

  it('isLeaf', () => {
    const withoutLoadData = render(
      <Tree showIcon={false} defaultExpandAll>
        <TreeNode isLeaf>
          <TreeNode isLeaf />
        </TreeNode>
      </Tree>
    );
    expect(renderToJson(withoutLoadData)).toMatchSnapshot();

    const withLoadData = render(
      <Tree defaultExpandAll>
        <TreeNode isLeaf>
          <TreeNode isLeaf />
        </TreeNode>
      </Tree>
    );
    expect(renderToJson(withLoadData)).toMatchSnapshot();

    const forceNoLeaf = render(
      <Tree defaultExpandAll>
        <TreeNode>
          <TreeNode isLeaf={false} />
        </TreeNode>
      </Tree>
    );
    expect(renderToJson(forceNoLeaf)).toMatchSnapshot();
  });

  describe('customize icon', () => {
    it('element', () => {
      const withoutLoadData = render(
        <Tree defaultExpandAll>
          <TreeNode icon={<span className="cust-icon" />} />
        </Tree>
      );
      expect(renderToJson(withoutLoadData)).toMatchSnapshot();
    });

    it('component', () => {
      const Icon = () => (
        <span className="cust-icon" />
      );

      const withoutLoadData = render(
        <Tree defaultExpandAll>
          <TreeNode icon={Icon} />
        </Tree>
      );
      expect(renderToJson(withoutLoadData)).toMatchSnapshot();
    });

    it('hide icon', () => {
      const withoutLoadData = render(
        <Tree showIcon={false} defaultExpandAll>
          <TreeNode icon={<span className="cust-icon" />} />
        </Tree>
      );
      expect(renderToJson(withoutLoadData)).toMatchSnapshot();
    });

    it('get props when loading', () => {
      const then = jest.fn(() => Promise.resolve());
      const loadData = jest.fn(() => ({ then }));
      const iconFn = jest.fn(() => null);
      const wrapper = mount(
        <Tree loadData={loadData}>
          <TreeNode icon={iconFn} title="parent 1" key="0-0" />
        </Tree>
      );

      wrapper.find('.rc-tree-switcher').simulate('click');

      expect(iconFn.mock.calls[0][0].loading).toBe(false);
      expect(iconFn.mock.calls[1][0].loading).toBe(true);
    });
  });

  describe('data and aria props', () => {
    it('renders data attributes on li', () => {
      const wrapper = render(
        <Tree defaultExpandAll>
          <TreeNode key="0-0" data-test="0-0">
            <TreeNode key="0-0-0" className="tree-node-cls" data-test="0-0-0">
              <TreeNode data-test="0-0-0-0"/>
            </TreeNode>
            <TreeNode key="0-0-1" data-test="0-0-1" />
          </TreeNode>
          <TreeNode key="0-1" data-test="0-1" />
        </Tree>
      );
      expect(renderToJson(wrapper)).toMatchSnapshot();
    });

    it('renders aria attributes on li', () => {
      const wrapper = render(
        <Tree defaultExpandAll>
          <TreeNode key="0-0" aria-label="0-0">
            <TreeNode key="0-0-0" className="tree-node-cls" aria-label="0-0-0">
              <TreeNode aria-label="0-0-0-0"/>
            </TreeNode>
            <TreeNode key="0-0-1" aria-label="0-0-1" />
          </TreeNode>
          <TreeNode key="0-1" aria-label="0-1" />
        </Tree>
      );
      expect(renderToJson(wrapper)).toMatchSnapshot();
    });
  });

  it('selectable', () => {
    const onClick = jest.fn();
    const onSelect = jest.fn();

    const wrapper = mount(
      <Tree selectable onClick={onClick} onSelect={onSelect}>
        <TreeNode selectable={false} />
      </Tree>
    );

    wrapper.find('.rc-tree-node-content-wrapper').simulate('click');

    expect(onClick).toBeCalled();
    expect(onSelect).not.toBeCalled();
  });
});
