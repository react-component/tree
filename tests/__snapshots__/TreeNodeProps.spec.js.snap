exports[`TreeNode Props className 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher_open">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__open">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
    <ul
      class="rc-tree-child-tree rc-tree-child-tree-open"
      data-expanded="true"
      role="group">
      <li
        class="tree-node-cls rc-tree-treenode-switcher-open"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher_open">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__open">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
        <ul
          class="rc-tree-child-tree rc-tree-child-tree-open"
          data-expanded="true"
          role="group">
          <li
            class="rc-tree-treenode-switcher-open"
            role="treeitem">
            <span
              class="rc-tree-switcher rc-tree-switcher-noop">
              
            </span>
            <span
              class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
              title="---">
              <span
                class="rc-tree-iconEle rc-tree-icon__docu">
                
              </span>
              <span
                class="rc-tree-title">
                ---
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="rc-tree-treenode-switcher-open"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher-noop">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__docu">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
      </li>
    </ul>
  </li>
  <li
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__docu">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
  </li>
</ul>
`;

exports[`TreeNode Props customize icon component 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__customize">
        <span
          class="cust-icon">
          
        </span>
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
  </li>
</ul>
`;

exports[`TreeNode Props customize icon element 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__customize">
        <span
          class="cust-icon">
          
        </span>
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
  </li>
</ul>
`;

exports[`TreeNode Props customize icon hide icon 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
  </li>
</ul>
`;

exports[`TreeNode Props data and aria props renders aria attributes on li 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    aria-label="0-0"
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher_open">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__open">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
    <ul
      class="rc-tree-child-tree rc-tree-child-tree-open"
      data-expanded="true"
      role="group">
      <li
        aria-label="0-0-0"
        class="tree-node-cls rc-tree-treenode-switcher-open"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher_open">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__open">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
        <ul
          class="rc-tree-child-tree rc-tree-child-tree-open"
          data-expanded="true"
          role="group">
          <li
            aria-label="0-0-0-0"
            class="rc-tree-treenode-switcher-open"
            role="treeitem">
            <span
              class="rc-tree-switcher rc-tree-switcher-noop">
              
            </span>
            <span
              class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
              title="---">
              <span
                class="rc-tree-iconEle rc-tree-icon__docu">
                
              </span>
              <span
                class="rc-tree-title">
                ---
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        aria-label="0-0-1"
        class="rc-tree-treenode-switcher-open"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher-noop">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__docu">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
      </li>
    </ul>
  </li>
  <li
    aria-label="0-1"
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__docu">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
  </li>
</ul>
`;

exports[`TreeNode Props data and aria props renders data attributes on li 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class="rc-tree-treenode-switcher-open"
    data-test="0-0"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher_open">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__open">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
    <ul
      class="rc-tree-child-tree rc-tree-child-tree-open"
      data-expanded="true"
      role="group">
      <li
        class="tree-node-cls rc-tree-treenode-switcher-open"
        data-test="0-0-0"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher_open">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__open">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
        <ul
          class="rc-tree-child-tree rc-tree-child-tree-open"
          data-expanded="true"
          role="group">
          <li
            class="rc-tree-treenode-switcher-open"
            data-test="0-0-0-0"
            role="treeitem">
            <span
              class="rc-tree-switcher rc-tree-switcher-noop">
              
            </span>
            <span
              class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
              title="---">
              <span
                class="rc-tree-iconEle rc-tree-icon__docu">
                
              </span>
              <span
                class="rc-tree-title">
                ---
              </span>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="rc-tree-treenode-switcher-open"
        data-test="0-0-1"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher-noop">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__docu">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
      </li>
    </ul>
  </li>
  <li
    class="rc-tree-treenode-switcher-open"
    data-test="0-1"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__docu">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
  </li>
</ul>
`;

exports[`TreeNode Props isLeaf 1`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class=""
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
    <ul
      class="rc-tree-child-tree rc-tree-child-tree-open"
      data-expanded="true"
      role="group">
      <li
        class=""
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher-noop">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
          title="---">
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`TreeNode Props isLeaf 2`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class=""
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher-noop">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__docu">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
    <ul
      class="rc-tree-child-tree rc-tree-child-tree-open"
      data-expanded="true"
      role="group">
      <li
        class=""
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher-noop">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-normal"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__docu">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`TreeNode Props isLeaf 3`] = `
<ul
  class="rc-tree"
  role="tree"
  unselectable="on">
  <li
    class="rc-tree-treenode-switcher-open"
    role="treeitem">
    <span
      class="rc-tree-switcher rc-tree-switcher_open">
      
    </span>
    <span
      class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
      title="---">
      <span
        class="rc-tree-iconEle rc-tree-icon__open">
        
      </span>
      <span
        class="rc-tree-title">
        ---
      </span>
    </span>
    <ul
      class="rc-tree-child-tree rc-tree-child-tree-open"
      data-expanded="true"
      role="group">
      <li
        class="rc-tree-treenode-switcher-open"
        role="treeitem">
        <span
          class="rc-tree-switcher rc-tree-switcher_open">
          
        </span>
        <span
          class="rc-tree-node-content-wrapper rc-tree-node-content-wrapper-open"
          title="---">
          <span
            class="rc-tree-iconEle rc-tree-icon__open">
            
          </span>
          <span
            class="rc-tree-title">
            ---
          </span>
        </span>
      </li>
    </ul>
  </li>
</ul>
`;
